python/lsst/daf/butler/registries/sqlRegistry.py

```diff
@@ -71,7 +98,28 @@ def getDatasetType(self, name):
         type : `DatasetType`
             The `DatasetType` associated with the given name.
         """
-        raise NotImplementedError("Must be implemented by subclass")
+        datasetType = None
+        if name in self._datasetTypes:
+            datasetType = self._datasetTypes[name]
+        else:
+            datasetTypeTable = self._schema.metadata.tables['DatasetType']
+            datasetTypeUnitsTable = self._schema.metadata.tables['DatasetTypeUnits']
+            with self._engine.begin() as connection:
+                # Get StorageClass and template from DatasetType table
+                result = connection.execute(select([datasetTypeTable.c.storage_class,
+                                                    datasetTypeTable.c.template]).where(
+                    datasetTypeTable.c.dataset_type_name == name)).fetchone()
+                storageClass = result['storage_class']
```

Comment:
But `DatasetType.storageClass` is documented to be a `StorageClass` not a `str`.

Reply:
I'll change the docstring. Should be a name (for now at least).